fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -3282,7 +3295,17 @@ public Bundle doHistory(MultivaluedMap<String, String> queryParameters,
             nextRequest.append(serviceBase);
             nextRequest.append("?");
             nextRequest.append("_count=").append(count);
-            nextRequest.append("&_afterHistoryId=").append(lastChangeId);
+
+            if (historyContext.getSince() != null && historyContext.getSince().getValue() != null) {
+                // As _since was given, we need to go with time-based paging, and the client
+                // will be responsible for managing duplicates.
+                nextRequest.append("&_since=").append(lastChangeTime.atZone(UTC).format(DateTime.PARSER_FORMATTER));
```

Comment:
not sure why we can't switch them from time-tased to resourceId-based paging, but I guess its probably fine.  in practice, i think it means that most people will just use the time-based paging.

Reply:
I'll try to explain in the doc update, but there's an issue with concurrency making switching problematic. I imagine that downstream systems will use the afterHistoryId, real people will use since.